preview_generator/preview/builder/document__scribus.py

```diff
@@ -29,32 +28,23 @@
 class DocumentPreviewBuilderScribus(DocumentPreviewBuilder):
     @classmethod
     def check_dependencies(cls) -> bool:
-        logger = logging.getLogger(LOGGER_NAME)
         try:
             # INFO - G.M - 2019-01-17 - stderr is redirected to devnull because
             # scribus print normal information to stderr instead of stdout.
-            check_output(["scribus", "-v"], stderr=STDOUT)
+            with Xvfb():
+                check_output(["scribus", "-v"], stderr=STDOUT)
             return True
+        except EnvironmentError as err:
+            raise BuilderDependencyNotFound(str(err))
         except FileNotFoundError:
             raise BuilderDependencyNotFound("this builder requires scribus to be available")
-        except CalledProcessError as exc:
-            # TODO - 2018/09/26 - Basile - using '-v' on scribus >= 1.5 gives
-            # the version then crash, using FileNotFoundError to make the diff
-            logger.warning(
-                "Scribus like missing (Note: scribus >= 1.5 can produce false error on this check): {}".format(
-                    exc.output
-                )
-            )
-            return True
 
     @classmethod
     def dependencies_versions(cls) -> typing.Optional[str]:
-        try:
+        with Xvfb():
```

Comment:
There is no more error handling there, did you see any case we can get dependencies_versions without running check_dependencies first ?

Reply:
This open the question for all implementations of `dependencies_versions`, most of them already don't implement this test, for example:
```
@classmethod
def dependencies_versions(cls) -> typing.Optional[str]:
    return "{} from {}".format(
        check_output(["inkscape", "--version"], universal_newlines=True).strip(),
        which("inkscape"),
    )
```

may raise a `CalledProcessError`.

The only call site of `dependencies_versions` is my `__main__.py`, which happen right after a check_dependencies.

Reply:
ok, thanks. Seems ok to me to not catch error here.